File: Python/面试经典150题/哈希表/49_字母异位词分组.py
from typing import List
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        map = {}
        ans = []
        for str in strs:
            sorted_s = ''.join(sorted(str))
            if sorted_s in map:
                map[sorted_s].append(str)
            else:
                map[sorted_s] = [str]
        for key,values in map.items():
            ans.append(values)
        return ans

strs = ["eat", "tea", "tan", "ate", "nat", "bat"]
print(Solution().groupAnagrams(strs))

File: JS/LeetCode热题HOT100/49_字母异位词分组.js
var groupAnagrams = function(strs) {
    let ans = []
    let map = new Map()
    for(let str of strs){
        let s = str.split("").sort().join("")
        if(map.has(s)){
            ans[map.get(s)].push(str) 
        }else{
            map.set(s, ans.length)
            ans.push([str])
        }
    }
    return ans
};